### DISCUSSION TITLE
Add donation bounties

### DISCUSSION THREAD

**Discussion by lwcorp** [sandboxie-plus/Sandboxie#1753] [Various ideas and suggestions]
Following #164, I'll be more specific - how about adding donation bounties instead of accepting general donations (with optional private notes)? See https://en.wikipedia.org/wiki/Open-source_bounty.

**Comment by DavidXanatos** [comment]
I don't know if that would work well, I'm under the impression that the majority of Sbie users just wants it to work as it did for the last decade or so.

Fixing compatibility bugs or worse security issues only after waiting long enough to reach a bounty does not sound like a very user friendly approach, as booth are somewhat time critical.

A general approach donate regularly to the project and I'll fix broken things as soon as I can seams more user friendly.

But for new features I could try something like that, do you have any particular platform in mind? 
I think with patreon I could only set goals like xxx@/month or so but not a enter xxx€, one time to get  something done.

**Comment by lwcorp** [comment]
Maybe one of the sites mentioned in the aforementioned Wikipedia link?

**Comment by Dyras** [comment]
Personally, I think what the Patreon needs is a short and concise description of what every tool does. I think that would be an incentive to sign up. As it is now, you kind of have to click around to see what they do.

Maybe under "About us"?

**Comment by HikariWS** [comment]
I've been using Sandboxie for many years.

Indeed the top priority is to enhance its compatibility as one of its primitives is to be transparent to all apps that run on a sandbox (as opposed to Windows Sandbox which requires apps to explicitly support it, which most probably requires some architecture overhaul, including split the app in at least 2 processes and a full review of what resources each process accesses).

It's much harder to support all kinds of resources handling and WinAPI calls than to just demand each software to be mostly rebuilt and blame them when something breaks while sandboxed lol

I've always had trouble running games on sandboxes, in example. Another need is to support Win7 and above features and APIs, as Sandboxie has barely had fixes on last years, for sure it doesn't support latest Windows stuff and I'm glad it just still works.

Of course new features are great. But without first fixing compatibility issues and bugs, new features just increase the chance of these issues getting worse and new issues arriving.

**Comment by isaak654** [comment]
In order to gain more visibility, I think GitHub Sponsors program could help: https://github.com/sponsors

**Comment by DavidXanatos** [comment]
Is there any simple platform suitable for this that is: put in a list of features, set out a minimum cost per feature and no upper limit. Such that the most demanded features can be done first.
The usual platforms I have seen are more guided towards integral projects, i.e. something with a basic goals and then some additional milestones. 
So its a linear progression scheme.
While here we want something parallel, with many goals and arbitrary order of implementation.

**Reply by lwcorp** [reply]
What about the list in [Wikipedia's Examples of websites listing bounties for multiple projects](https://en.wikipedia.org/w/index.php?title=Open-source_bounty&oldid=1092591003#Examples_of_websites_listing_bounties_for_multiple_projects)? It's mentioned one of them even "has GitHub integration and allows OSS maintainers to add bounties to specific issues on their GitHub repositories, and award contributors for pull requests that solve the issue."

_- link updated_

**Reply by isaak654** [reply]
Confirmed, I also shared some on the Slack channel.

**Reply by lwcorp** [reply]
By Slack I guess you mean #479 - just mentioning it since I don't think it's mentioned anywhere else (maybe it should).

**Comment by isaak654** [comment]
I think it's becoming necessary to consider a bug bounty system for any test subject that would grant a proper remote access to @DavidXanatos in order to fix major issues like BSODs in exchange of a [huge supporter certificate](https://xanasoft.com/product/sandboxie-plus-huge-supporter-certificate/) or money that could be compensated with the platforms where devs get paid for new features and new [sponsorships](https://github.com/sandboxie-plus/Sandboxie/discussions/1753#discussioncomment-2492126).

**Comment by volvoplz** [comment]
The issue I can see about the bounty is how it translates in term of mindset. 

When you give regularly on Patreon, you understand that you might receive more support than most users, but you also understand that this extra support is merely a potential perk, and not really the point of donating. The point of donating regularly is to give visibility and incentive to the maintainer(s) to ensure the long-term viability of the project, and reducing the risk of the software being abandoned. And in the case of sandboxie, as we all know the history, it makes a lot of sense. I was a paying customer 5 years ago, but knew the 30 euros/computer was not enough money for a company to maintain a niche software like that. It was kind of obvious that sandboxie would die if it wasn't supported by a subscription model.

In the current context, people are not paying for new features or even for support. At least I don't. As a "large-supporter", whatever that is, I don't feel entitled to support and even less to bug fixed or new features. I just feel really grateful and lucky the project is maintained, and I want to give regularly to show appreciation and give visibility/incentive to the devs so they keep maintaining it.

As personal users, many people would consider 100 euros or 100 USD a large some of money in the context. If they were to give that kind of money, they would have very high expectations. Many people would ask for bug being fixed promptly, or huge new features in exchange for "such" a bounty, without realizing that 100 euros would be equivalent to very few billable hours for a software engineer. 
The high expectations on the users(slash customers) and the low incentive for the developers would definitely create tensions. 

TL:DR : Bounties would make people feel entitled, in spite of the money being to low to be a real incentive for the devs. On the opposite, patreon/donation is a model that make people feel grateful without giving them the impression they are owed anything. They give to support, not to "buy" or be owed anything in return. Less pressure to the devs, and more solid/stable/revenues. The community also benefits more from devs that can focus on long term, consistent software because they see solid patreon revenues, instead of devs working on inconsistent/niche demands to fulfill bounties.

**Reply by lwcorp** [reply]
I understand where you come from, but no one asked to replace "true" donations. It's just an additional type of incentive.
Endless open source softwares do it (possibly while accepting regular donations in parallel) and it just "works". Why should Sandboxie be the exception to the rule?

Plus I think bounties work the other way around. The developers get the money only if the bountie is successful. Kind of like Kickstarter.

**Comment by HikariWS** [comment]
It's just a matter of him to define a list of stuff he requires money to do and how much. Of course it doesn't work if ppl offer whatever they want and even not knowing how the money will be spent.

I think this idea came to fund buying the certificate to sign the driver.
